### PR TITLE
Add unit test TestPruneDisconnectedRevTreeWithLongWinningBranch()

### PR DESCRIPTION
**Pull request by tleyden**
Verifies certain rev tree pruning behavior with disconnected branches.  Test passes.

### CODE REVIEW

**Comment by tleyden** [comment]
Here are the test steps

## Initial Rev Tree

testprunedisconnectedrevtreewithlongwinningbranch_initial dot


## Prune to maxdepth=7

testprunedisconnectedrevtreewithlongwinningbranch_pruned1 dot


## Add more revs to winning branch

testprunedisconnectedrevtreewithlongwinningbranch_add_winning_revs dot


## Prune (again) to maxdepth=7

testprunedisconnectedrevtreewithlongwinningbranch_pruned_final dot

## Notes on pruning result

* The winning branch is correctly pruned down to 7 revisions (this is the behavior I wanted to verify with the new unit test)
* The non-winning branch is not affected

File: db/revtree_test.go [hunk 1]
```diff
@@ -602,6 +603,62 @@ func TestLongestBranch2(t *testing.T) {
...
+		ioutil.WriteFile("/tmp/TestPruneDisconnectedRevTreeWithLongWinningBranch_add_winning_revs.dot", []byte(revTree.RenderGraphvizDot()), 0666)
+	}
+
+	revTree.pruneRevisions(maxDepth, "")
+
+	if (dumpRevTreeDotFiles) {
+		ioutil.WriteFile("/tmp/TestPruneDisconnectedRevTreeWithLongWinningBranch_pruned_final.dot", []byte(revTree.RenderGraphvizDot()), 0666)
+	}
+
+	// Make sure the winning branch is pruned down to 20, even with the disconnected rev tree
```
Comment (JFlath) [inline db/revtree_test.go]: Nit: comment didn't get changed to `7` between your commits

Comment (tleyden) [reply db/revtree_test.go]: Thanks!  fixed

**Comment by tleyden** [comment]
---------------------------------------------------------
---------------------------------------------------------

## Test against SG 1.4.x 

I backported the test in this PR to the [feature/pr_2714_verify_rev_tree_pruning_sg_14](https://github.com/couchbase/sync_gateway/tree/feature/pr_2714_verify_rev_tree_pruning_sg_14) branch, and ran the test.  Here are the results.

## Initial Rev Tree

testprunedisconnectedrevtreewithlongwinningbranch_initial dot

## Prune to maxdepth=7

testprunedisconnectedrevtreewithlongwinningbranch_pruned1 dot

**NOTE: already the pruning behavior has diverged from post #2651 fixed version**

## Add more revs to winning branch

testprunedisconnectedrevtreewithlongwinningbranch_add_winning_revs dot


## Prune (again) to maxdepth=7

testprunedisconnectedrevtreewithlongwinningbranch_pruned_final dot

## Notes on pruning result

- Less pruning happening than the post #2651 fixed version tested in https://github.com/couchbase/sync_gateway/pull/2714#issuecomment-314910806

**Comment by tleyden** [comment]
Also, if the test is modified to add more revs to the winning branch:

```
        // Add revs to winning branch
	addRevs(
                revTree,
                winningBranchStartRev,
		50,
                "winning",
	)
```

It appears that the winning branch will grow without bound in this case:

See [TestPruneDisconnectedRevTreeWithLongWinningBranch_pruned_final.dot.png](http://cbmobile-bucket.s3.amazonaws.com/diagrams/example-sync-gateway-revtrees/TestPruneDisconnectedRevTreeWithLongWinningBranch_pruned_final.dot.png)

**Pull request merged**